cmd/api-linter/cli.go

```diff
@@ -132,29 +136,33 @@ func (c *cli) lint(rules lint.RuleRegistry, configs lint.Configs) error {
 	// Resolve file absolute paths to relative ones.
 	protoFiles, err := protoparse.ResolveFilenames(c.ProtoImportPaths, c.ProtoFiles...)
 	if err != nil {
-		return err
+		return lintFailure, err
 	}
 	fd, err := p.ParseFiles(protoFiles...)
 	if err != nil {
 		if err == protoparse.ErrInvalidSource {
 			if len(errorsWithPos) == 0 {
-				return errors.New("got protoparse.ErrInvalidSource but no ErrorWithPos errors")
+				return lintFailure, errors.New("got protoparse.ErrInvalidSource but no ErrorWithPos errors")
 			}
 			// TODO: There's multiple ways to deal with this but this prints all the errors at least
 			errStrings := make([]string, len(errorsWithPos))
 			for i, errorWithPos := range errorsWithPos {
 				errStrings[i] = errorWithPos.Error()
 			}
-			return errors.New(strings.Join(errStrings, "\n"))
+			return lintFailure, errors.New(strings.Join(errStrings, "\n"))
 		}
-		return err
+		return lintFailure, err
 	}
 
 	// Create a linter to lint the file descriptors.
 	l := lint.New(rules, configs)
 	results, err := l.LintProtos(fd...)
 	if err != nil {
-		return err
+		return lintFailure, err
+	}
+
+	if c.ExitStatusOnLintFailure && len(results) > 0 {
+		lintFailure = true
```

Comment:
I think we can return an error, like
```go
if c.ExitStatusOnLintFailure && len(results) > 0 {
  return fmt.Errorf("Found %d linting results", len(results))
}
```
And in the `main` function, `Fatalf` is equivalent to `Printf()` followed by a call to `os.Exit(1)`. And you can remove `lintFailure`.

Reply:
Thanks for the suggestion, it looks a lot cleaner now.